app/js/app.js

```diff
@@ -7,129 +7,80 @@ pw.config(function($mdThemingProvider) {
 
 pw.run(['$rootScope', function($rootScope){
 
-	$rootScope.fullscreen = false;
-	$rootScope.initialized = false;
-  $rootScope.mapInitialized = false;
+  $rootScope.fullscreen = false;
+  $rootScope.initialized = false;
+
+  llb_app.addListener('window_state', function(data){
+    if(data.fullscreen)
+    {
+      $rootScope.$apply(function(){
+        $rootScope.fullscreen = true;
+      })
+    }
+    else
+    {
+      $rootScope.$apply(function(){
+        $rootScope.fullscreen = false
+      })
+    }
+  })
 
-	llb_app.addListener('window_state', function(data){
-		if(data.fullscreen)
-		{
-			$rootScope.$apply(function(){
-				$rootScope.fullscreen = true;
-        setMapSize($rootScope.window_dimensions.fullscreen_width, $rootScope.window_dimensions.fullscreen_height);
+  llb_app.request('window_dimensions')
+  llb_app.addListener('window_dimensions', function(data){
+    $rootScope.$apply(function(){
+      $rootScope.window_dimensions = data
+      $rootScope.fullscreen_app_dimensions = {
+        "width": data.fullscreen_width + "px",
+        "height": data.fullscreen_height - 64 + "px"
```

Comment:
In my opinion the data that the platform gives us should already have the toolbar height removed from the fullscreen_height, as we can't use that space anyhow anyway.